transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/CreateTransferFragment.java

```diff
@@ -0,0 +1,379 @@
+                //TODO temporary action for transfer destination clicked; it should open up add transfer method UI
+                Snackbar.make(mAddTransferDestination, "ADD Transfer clicked", Snackbar.LENGTH_SHORT).show();
+            }
+        });
+
+        // toggle button
+        mTransferAllSwitch = view.findViewById(R.id.switchButton);
+        mTransferAllSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                mCreateTransferViewModel.setTransferAllAvailableFunds(isChecked);
+            }
+        });
+        registerObserver();
+    }
+
+    private void prepareTransferAmount() {
+        mTransferAmount.setImeOptions(EditorInfo.IME_FLAG_NO_EXTRACT_UI | EditorInfo.IME_ACTION_NEXT);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            mTransferAmount.setImportantForAutofill(View.IMPORTANT_FOR_AUTOFILL_NO);
+        }
+        mTransferAmount.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
+                    mCreateTransferViewModel.setTransferAmount(((EditText) v).getText().toString());
+                }
+            }
+        });
+        mTransferAmount.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if (before != count) {
+                    mCreateTransferViewModel.setTransferAmount(s.toString());
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+    }
+
+    private void prepareTransferNotes() {
+        mTransferNotes.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
+                    mCreateTransferViewModel.setTransferNotes(((EditText) v).getText().toString());
+                }
+            }
+        });
+        mTransferNotes.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if (before != count) {
+                    mCreateTransferViewModel.setTransferNotes(s.toString());
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+    }
+
+    private void registerObserver() {
```

Comment:
Please take a look at the documentation. This method should be split into 3, registerTransferDestinationObserver, registerAvailableFundsObserver and registerTransferAmountObserver